src/cdk/tree/tree.ts

```diff
@@ -214,7 +214,9 @@ export class CdkTree<T> implements CollectionViewer, OnInit, OnDestroy {
     // Set the data to just created `CdkTreeNode`.
     // The `CdkTreeNode` created from `createEmbeddedView` will be saved in static variable
     //     `mostRecentTreeNode`. We get it from static variable and pass the node data to it.
-    CdkTreeNode.mostRecentTreeNode.data = nodeData;
+    if (CdkTreeNode.mostRecentTreeNode) {
```

Comment:
Under what circumstances would this be undefined?

Reply:
We have to initialize `mostRecentTreeNode` because of the error ` /usr/local/google/home/tinagao/WebstormProjects/material2/src/cdk/tree/node.ts:79:10: Metadata collected contains an error that will be reported at runtime: Only initialized variables and constants can be referenced because the value of this variable is needed by the template compiler.`

So I initialize the static variable to be null, and we have to check whether it's null here.

But the value shouldn't be `null` or `undefined` when we call it from `tree.ts`